File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int curmin = prices[0], answer = 0;
        for (int i = 1; i < prices.size(); i++) {
            answer = max(answer, prices[i] - curmin);
            curmin = min(curmin, prices[i]);
        }
        return answer;
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        
        minBuy = 987654321
        maxProfit = 0
        
        for price in prices:
            
            if price < minBuy:
                minBuy = price
                continue
            
            maxProfit = maxProfit if price - minBuy < maxProfit else price - minBuy
                
        return maxProfit
            